### DISCUSSION TITLE
How to sync custom uiState with custom connector state

### DISCUSSION THREAD

**Discussion by jsdmc** [algolia/instantsearch#5345] [Q&A] [answered]
HI guys.
I'm was looking for a solution how to add a custom prop to `indexUiState` and found this discussion https://github.com/algolia/react-instantsearch/discussions/3574
As I understood from @Haroenv comments, the only way to keep custom prop in state is to add widget that will take care of it.

```
export type ExtendedIndexUiState = IndexUiState & {
  customProp?: string; 
}
....
const { setIndexUiState } = useInstantSearch<ExtendedIndexUiState>();
setIndexUiState({ customProp: 'test' });
```
The problem is that after calling of `setIndexUiState` somewhere in the app, widget doesn't  get updated `uiState` in `getWidgetUiState`.

```
getWidgetUiState(uiState) {
  return uiState;
}
```
My question is: "How to sync custom uiState prop that was set outside widget with widget state?"
Thank you

**Comment by Haroenv** [comment, marked as answer]
Do you have a example of what you've tried so far in a sandbox? it will be much easier to debug that way. Thanks!

**Reply by jsdmc** [reply]
@Haroenv Thanks. Already figured out what I missed. I needed to read uiState values from second param in `getWidgetSearchParameters`. This way inner widget state can be synced with uiState that changed from outside.
Something like that:
```
getWidgetSearchParameters(searchParams, { uiState }) {
    widgetState.customProp = uiState.customProp;
    ...
}
```

**Reply by Haroenv** [reply]
That's right, getWidgetSearchParameters needs to sync with getWidgetUiState two-way :) those functions should be the opposite of eachother
